spec/jobs/reports/idv_legacy_conversion_supplement_report_spec.rb

```diff
@@ -0,0 +1,333 @@
+            end_date: iaa3_range.end,
+          )
+        end
+
+        let(:iaa3) { 'iaa3' }
+
+        let(:iaa3_sp1) do
+          create(
+            :service_provider,
+            iaa: iaa3,
+            iaa_start_date: iaa3_range.begin,
+            iaa_end_date: iaa3_range.end,
+          )
+        end
+
+        let(:iaa_order3) do
+          build_iaa_order(order_number: 3, date_range: iaa3_range, iaa_gtc: gtc3)
+        end
+
+        let(:iaa_order_expired) do
+          build_iaa_order(order_number: 2, date_range: expired_iaa_range, iaa_gtc: gtc3)
+        end
+
+        let(:integration_1) do
+          build_integration(
+            issuer: iaa3_sp1.issuer,
+            partner_account: partner_account3,
+          )
+        end
+
+        let(:user1) { create(:user) }
+
+        let(:user2) { create(:user) }
+
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
+
+        before do
+          iaa_order3.integrations << integration_1
+
+          create(
+            :sp_upgraded_biometric_profile,
+            issuer: iaa3_sp1.issuer, user_id: user1.id, upgraded_at: iaa3_range.begin + 1.day,
+          )
+          create(
+            :sp_upgraded_biometric_profile,
+            issuer: iaa3_sp1.issuer, user_id: user2.id, upgraded_at: iaa3_range.begin + 1.month,
+          )
+        end
+
+        it 'finds data for the same issuer with different year_months' do
+          expect(csv.length).to eq(2)
+          aggregate_failures do
+            row = csv.find { |r| r['issuer'] == iaa3_sp1.issuer && r['year_month'] == '202009' }
+
+            expect(row.to_h.symbolize_keys).to eq(
+              {
+                iaa_order_number: 'gtc9101-0003',
+                iaa_start_date: '2020-09-01',
+                iaa_end_date: '2021-08-30',
+                issuer: iaa3_sp1.issuer,
+                friendly_name: iaa3_sp1.friendly_name,
+                year_month: '202009',
+                year_month_readable: 'September 2020',
+                user_count: '1',
+              },
+            )
+          end
+
+          aggregate_failures do
+            row = csv.find { |r| r['issuer'] == iaa3_sp1.issuer && r['year_month'] == '202010' }
+
+            expect(row.to_h.symbolize_keys).to eq(
+              {
+                iaa_order_number: 'gtc9101-0003',
+                iaa_start_date: '2020-09-01',
+                iaa_end_date: '2021-08-30',
+                issuer: iaa3_sp1.issuer,
+                friendly_name: iaa3_sp1.friendly_name,
+                year_month: '202010',
+                year_month_readable: 'October 2020',
+                user_count: '1',
+              },
+            )
+          end
+        end
+
+        context 'when there is an expired iaa with the same issuer' do
+          before do
+            iaa_order_expired.integrations << integration_1
+          end
+
+          it 'does not include the expired iaa in the report' do
```

Comment:
This is kind of a silly test. But it shows that by adding an older order with an issuer, the logic works. I could move the before to the test above but this is more explicit.